### PR TITLE
DOCS-#2971: Document `modin.distributed.dataframe.pandas`

### LINKED ISSUE #2971: Document `modin.distributed.dataframe.pandas`

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2971  
- [x] tests passing

### CODE REVIEW

**Review by vnlitvinov — DISMISSED** [review]

File: docs/developer/partition_api.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Partition API in Modin
 ======================
 
-When you are working with a Modin Dataframe, you can unwrap its remote partitions
+When you are working with a Modin DataFrame, you can unwrap its remote partitions
```
Comment (devin-petersohn) [inline docs/developer/partition_api.rst:4]: I prefer to revert this change. DataFrame is camel case  because of "data frame". I prefer "dataframe".

File: docs/developer/partition_api.rst [hunk 2]
```diff
@@ -11,34 +11,15 @@ Partition IPs
 For finer grained placement control, Modin also provides an API to get the IP addresses of the nodes that hold each partition.
 You can pass the partitions having needed IPs to your function. It can help with minimazing of data movement between nodes.
 
-unwrap_partitions
------------------
+Partition API implementations
+-----------------------------
+Currently, a Modin DataFrame stores underlying partitions as pandas DataFrame(s).
```
Comment (devin-petersohn) [inline docs/developer/partition_api.rst:16]:
```suggestion
By default, a Modin Dataframe stores underlying partitions as ``pandas.DataFrame`` objects.
```

File: docs/developer/partition_api.rst [hunk 3]
```diff
@@ -11,34 +11,15 @@ Partition IPs
...
+You can find conforming API implementation in :doc:`Pandas Partition API </flow/modin/distributed/dataframe/pandas>`.
```
Comment (devin-petersohn) [inline docs/developer/partition_api.rst:17]:
```suggestion
You can find the specific implementation of Modin's Partition Interface in :doc:`Pandas Partition API </flow/modin/distributed/dataframe/pandas>`.
```

File: docs/flow/modin/distributed/dataframe/pandas.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
+Pandas Partition API
+====================
+
+This page contains description of API to operate on Modin DataFrame partitions that are pandas DataFrame(s).
```
Comment (devin-petersohn) [inline docs/flow/modin/distributed/dataframe/pandas.rst:4]:
```suggestion
This page contains a description of the API to extract partitions from and build Modin Dataframes.
```

**Review by devin-petersohn — COMMENTED** [review]
Thanks @YarShev, left a couple of comments.

File: docs/developer/partition_api.rst (hunk 1 above)
Comment (YarShev) [reply docs/developer/partition_api.rst:4]: Okay, I do not mind. Probably, we will need to fix `DataFrame` -> `dataframe` in many places with one commit after we complete documentation changes :)

**Review by devin-petersohn — APPROVED** [review]

File: docs/developer/partition_api.rst (hunk 1 above)
Comment (vnlitvinov) [reply docs/developer/partition_api.rst]: Created https://github.com/modin-project/modin/issues/3070 to track.
